content/docs/hooks-reference.md

```diff
@@ -25,29 +25,29 @@ If you're new to Hooks, you might want to check out [the overview](/docs/hooks-o
   - [`useLayoutEffect`](#uselayouteffect)
   - [`useDebugValue`](#usedebugvalue)
 
-## Basic Hooks {#basic-hooks}
+## الخطافات الأساسية {#basic-hooks}
 
 ### `useState` {#usestate}
 
 ```js
 const [state, setState] = useState(initialState);
 ```
 
-Returns a stateful value, and a function to update it.
+يعيد هذا الخطاف قيمةً ذات حالة، ودالةً لتحديث هذه القيمة.
 
-During the initial render, the returned state (`state`) is the same as the value passed as the first argument (`initialState`).
+أثناء عملية التصيير الأولية، الحالة المعادة (`state`) هي نفسها القيمة المُمرَّرة كأول وسيط (`initialState`).
 
-The `setState` function is used to update the state. It accepts a new state value and enqueues a re-render of the component.
+تُستعمَل الدالة `setState` لتحديث الحالة، إذ تقبل قيمةً جديدةً للحالة وتدرج في الطابور عملية إعادة تصيير لمكون.
 
 ```js
 setState(newState);
 ```
 
-During subsequent re-renders, the first value returned by `useState` will always be the most recent state after applying updates.
+أثناء عمليات إعادة التصيير اللاحقة، القيمة الأولى التي يعيدها الخطاف `useState` ستبقى دومًا أحدث حالة بعد تطبيق التحديثات.
 
```

Comment:
Please rephrase the translation for this part because it is unclear .. 
![image](https://user-images.githubusercontent.com/43971542/61093469-f9faa900-a44a-11e9-835e-0b862ab24ba2.png)

Reply:
Can you provide a better translation please

Reply:
يضمن React أن هوية دالة `setState` مستقرة ولن تتغير في عمليات إعادة التصيير. هذا هو السبب في أنه من الآمن حذفها من قائمة التبعية `useEffect` أو `useCallback` .

![text-clear-to-read](https://user-images.githubusercontent.com/43971542/61172487-637ed280-a585-11e9-9efd-75e281017296.png)